mbuild/conversion.py

```diff
@@ -716,6 +774,66 @@ def from_pybel(pybel_mol,
     return compound
 
 
+def from_rdkit(rdkit_mol,
+               compound=None,
+               coords_only=False,
+               smiles_seed=0):
+    """Return an mbuild Compound based on a smiles string using RDKit.
+
+    Parameters
+    ---------
+    rdkit_mol : rdkit.Chem.rdchem.Mol
+        RDKit mol to generate an mBuild compound
+    compound : mb.Compound, optional, default=None
+        The host mbuild Compound.
+    coords_only : bool, optional, default=False
+        Set preexisting atoms in compound to coordinates given
+        by structure. Note: Not yet implemented, included only
+        for parity with other conversion functions.
+    smiles_seed : int, optional, default=0
+        Random number seed for PRNG, set to -1 for non-deterministic behavior
+
+    Returns
+    -------
+    mbuild.Compound
+
+    NOTES
+    -----
+    Option `coords_only` currently is not implemented, it is only provided to
+        maintain parity with other conversion methods.
+
+    """
+    from rdkit import Chem
+    from rdkit.Chem import AllChem
+
+    mymol = Chem.AddHs(rdkit_mol)
+    AllChem.EmbedMolecule(mymol, randomSeed=smiles_seed)
```

Comment:
This can fail for some molecules. E.g., try the `PF6` anion. IIRC, it returns `-1` if it fails and `0` otherwise. We probably should add some error handling and suggest that people try `openbabel` if the SMILES generation fails.

Reply:
Agreed, will add in a catch for this.

Reply:
A check has been added, as well as a unit test to ensure the error is properly raised.